### ISSUE TITLE
Api::patch should only allow modifications to a subset type

### ISSUE THREAD

**Issue by clux** [kube-rs/kube-rs#175]
This can be done using some ideas from #143. However, it is made complicated by the json patch format that sits on top of it.

This is why we should try to do it for `Api::apply` #174 as a starting point (which is likely what most future controllers will use in the future any because patch can be difficult to use directly).

**Comment by clux** [comment]
I think this idea is probably a bit too ambitious, particularly when seeing how brittle the serialization can be when looking back at the recent k8s-openapi option unpacking experiment.

Going to close this issue. I don't think we have a clear way to do this, nor is it necessary to do this. If that changes, we can open a new issue.

**Comment by clux** [comment]
Well, an analogue of this was actually done in `client-go` now under [KEP-2155 :: generated apply configuration types](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/2155-clientgo-apply#generated-apply-configuration-types), so it might be worth re-examining if we want to pick up #649. But will wait on discussion therein before re-opening this (particularly with other work on protobuf serialization).
